### PR TITLE
Add RU calcuation to explain docs

### PR DESCRIPTION
**Pull request by adamf-db**
### What is changed, added or deleted? (Required)

Added details on how RUs are calculated to the `EXPLAIN ANALZYE` documentation. See https://github.com/pingcap/tidb/issues/47269

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [X] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-explain-analyze.md [hunk 1]
```diff
@@ -294,6 +294,32 @@ RU:273.842670
...
+The RU count can be calculated from the other values in `EXPLAIN ANALYZE`; specifically, the `executeInfo` block. For example, given:
+
+```
+ 'executeInfo': 'time:2.55ms, loops:2, RU:0.329460, Get:{num_rpc:1, '
+                'total_time:2.13ms}, total_process_time: 231.5µs, '
+                'total_wait_time: 732.9µs, tikv_wall_time: 995.8µs, '
+                'scan_detail: {total_process_keys: 1, total_process_keys_size: '
+                '150, total_keys: 1, get_snapshot_time: 691.7µs, rocksdb: '
+                '{block: {cache_hit_count: 2, read_count: 1, read_byte: 8.19 '
+                'KB, read_time: 10.3µs}}}',
```
Comment (dveeden) [inline sql-statements/sql-statement-explain-analyze.md:306]: Maybe we can format this a bit more:
```suggestion
 'executeInfo':
     time:2.55ms, 
     loops:2, 
     RU:0.329460, 
     Get:{
         num_rpc:1,
         total_time:2.13ms
     }, 
     total_process_time: 231.5µs,
     total_wait_time: 732.9µs, 
     tikv_wall_time: 995.8µs,
     scan_detail: {
        total_process_keys: 1, 
        total_process_keys_size: 150, 
        total_keys: 1, 
        get_snapshot_time: 691.7µs,
        rocksdb: {
            block: {
                cache_hit_count: 2,
                read_count: 1,
                read_byte: 8.19 KB,
                read_time: 10.3µs
            }
        }
    },
```

File: sql-statements/sql-statement-explain-analyze.md [hunk 2]
```diff
@@ -294,6 +294,32 @@ RU:273.842670
...
+```
+
+The base costs are in [the pd source](https://github.com/tikv/pd/blob/aeb259335644d65a97285d7e62b38e7e43c6ddca/client/resource_group/controller/config.go#L58C19-L67) and the calculations are done in [model.go](https://github.com/tikv/pd/blob/54219d649fb4c8834cd94362a63988f3c074d33e/client/resource_group/controller/model.go#L107)
+
+If you are using 7.1.x, the calculation is sum `BeforeKVRequest() + AfterKVRequest()` in `pd/pd-client/model.go`, that is:
```
Comment (dveeden) [inline sql-statements/sql-statement-explain-analyze.md:311]: Should this be v7.1 and newer (or older)? Or specifically v7.1?

File: sql-statements/sql-statement-explain-analyze.md [hunk 3]
```diff
@@ -294,6 +294,32 @@ RU:273.842670
...
+
+```
+before :
```
Comment (dveeden) [inline sql-statements/sql-statement-explain-analyze.md:314]: before what? before v7.1?

**Review by dveeden — APPROVED** [review]

File: sql-statements/sql-statement-explain-analyze.md (hunk 3 above)
Comment (adamf-db) [reply sql-statements/sql-statement-explain-analyze.md]: Updated to make it more clear that it means before request is processed/after request is processed.

File: sql-statements/sql-statement-explain-analyze.md (hunk 2 above)
Comment (adamf-db) [reply sql-statements/sql-statement-explain-analyze.md]: Updated to 7.1+

**Comment by adamf-db** [comment]
@dcalvin can you review?

**Comment by adamf-db** [comment]
Ping on this for a 2nd approver.

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
ptal @hfxsd

File: sql-statements/sql-statement-explain-analyze.md [hunk 4]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+```
+before key/value request is processed:
+      consumption.RRU += float64(kc.ReadBaseCost) -> kv.ReadBaseCost * rpc_nums
+
+after key/value request is processed:
+      consumption.RRU += float64(kc.ReadBytesCost) * readBytes -> kc.ReadBytesCost * total_process_keys_size
+      consumption.RRU += float64(kc.CPUMsCost) * kvCPUMs -> kc.CPUMsCost * total_process_time
+```
+
+For writes & batch gets the calculation is similar with different base values.
```
Comment (hfxsd) [inline sql-statements/sql-statement-explain-analyze.md:340]:
```suggestion
For writes and `BatchGet`, the calculation is similar with different base values.
```

**Review by hfxsd — APPROVED** [review]

File: sql-statements/sql-statement-explain-analyze.md [hunk 5]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+                read_byte: 8.19 KB,
+                read_time: 10.3µs
+            }
+        }
+    },
+```
+
+The base costs are in [the pd source](https://github.com/tikv/pd/blob/aeb259335644d65a97285d7e62b38e7e43c6ddca/client/resource_group/controller/config.go#L58C19-L67) and the calculations are done in [model.go](https://github.com/tikv/pd/blob/54219d649fb4c8834cd94362a63988f3c074d33e/client/resource_group/controller/model.go#L107)
+
+If you are using 7.1+, the calculation is sum `BeforeKVRequest() + AfterKVRequest()` in `pd/pd-client/model.go`, that is:
```
Comment (hfxsd) [inline sql-statements/sql-statement-explain-analyze.md:329]:
```suggestion
If you are using TiDB v7.1, the calculation is sum `BeforeKVRequest() + AfterKVRequest()` in `pd/pd-client/model.go`, that is:
```

File: sql-statements/sql-statement-explain-analyze.md [hunk 6]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
 >
 > This value shows the actual RUs consumed by this execution. The same SQL statement might consume different amounts of RUs each time it is executed due to the effects of caching (for example, [coprocessor cache](/coprocessor-cache.md)).
 
+The RU count can be calculated from the other values in `EXPLAIN ANALYZE`; specifically, the `executeInfo` block. For example, given:
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain-analyze.md:297]:
```suggestion
You can calculate the RU from the other values in `EXPLAIN ANALYZE`, specifically the `execution info` column. For example:
```

File: sql-statements/sql-statement-explain-analyze.md [hunk 7]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+        rocksdb: {
+            block: {
+                cache_hit_count: 2,
+                read_count: 1,
+                read_byte: 8.19 KB,
+                read_time: 10.3µs
+            }
+        }
+    },
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain-analyze.md:325]:
````suggestion
```json
'executeInfo':
   time:2.55ms, 
   loops:2, 
   RU:0.329460, 
   Get:{
       num_rpc:1,
       total_time:2.13ms
   }, 
   total_process_time: 231.5µs,
   total_wait_time: 732.9µs, 
   tikv_wall_time: 995.8µs,
   scan_detail: {
      total_process_keys: 1, 
      total_process_keys_size: 150, 
      total_keys: 1, 
      get_snapshot_time: 691.7µs,
      rocksdb: {
          block: {
              cache_hit_count: 2,
              read_count: 1,
              read_byte: 8.19 KB,
              read_time: 10.3µs
          }
      }
  },
```
````

File: sql-statements/sql-statement-explain-analyze.md [hunk 8]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+
+The base costs are in [the pd source](https://github.com/tikv/pd/blob/aeb259335644d65a97285d7e62b38e7e43c6ddca/client/resource_group/controller/config.go#L58C19-L67) and the calculations are done in [model.go](https://github.com/tikv/pd/blob/54219d649fb4c8834cd94362a63988f3c074d33e/client/resource_group/controller/model.go#L107)
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain-analyze.md:327]:
```suggestion
The base costs are defined in the [`tikv/pd` source code](https://github.com/tikv/pd/blob/aeb259335644d65a97285d7e62b38e7e43c6ddca/client/resource_group/controller/config.go#L58C19-L67) and the calculations are performed in the [`model.go`](https://github.com/tikv/pd/blob/54219d649fb4c8834cd94362a63988f3c074d33e/client/resource_group/controller/model.go#L107) file.
```

File: sql-statements/sql-statement-explain-analyze.md [hunk 9]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+
+If you are using TiDB v7.1, the calculation is sum `BeforeKVRequest() + AfterKVRequest()` in `pd/pd-client/model.go`, that is:
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain-analyze.md:329]:
```suggestion
If you are using TiDB v7.1, the calculation is the sum of `BeforeKVRequest()` and `AfterKVRequest()` in `pd/pd-client/model.go`, that is:
```

File: sql-statements/sql-statement-explain-analyze.md [hunk 10]
```diff
@@ -294,6 +294,51 @@ RU:273.842670
...
+```
+before key/value request is processed:
+      consumption.RRU += float64(kc.ReadBaseCost) -> kv.ReadBaseCost * rpc_nums
+
+after key/value request is processed:
+      consumption.RRU += float64(kc.ReadBytesCost) * readBytes -> kc.ReadBytesCost * total_process_keys_size
+      consumption.RRU += float64(kc.CPUMsCost) * kvCPUMs -> kc.CPUMsCost * total_process_time
+```
+
+For writes and batch gets, the calculation is similar with different base values.
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain-analyze.md:340]:
```suggestion
For writes and batch gets, the calculation is similar with different base costs.
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**